src/rt-app_taskgroups.c

```diff
@@ -373,7 +435,35 @@ static int cgroup_attach_task(char *name)
 	FILE *tasks;
 	size_t size;
 
-	file = strcmp(name, "/") ? "/tasks" : "tasks";
+	if (ctrl.version == 1) {
+		file = strcmp(name, "/") ? "/tasks" : "tasks";
+	}
+
+	if (ctrl.version == 2) {
+		if (strcmp(name, "/")) {
+			size_t path_size = strlen(ctrl.mount_point) + strlen(name) + strlen(TYPE) + 3;
+			char* cgroup_path = malloc(path_size);
+			FILE* cgroup_file;
+
+			snprintf(cgroup_path, path_size, "%s/%s/%s", ctrl.mount_point, name, TYPE);
+			cgroup_file = fopen(cgroup_path, "we");
+			if (!cgroup_file) {
+				perror("open");
+				goto error;
+			}	
+			if (fprintf(cgroup_file, "threaded") < 0) {
+				perror("fprintf");
```

Comment:
cgroup_file not closed here

Reply:
Also some better message may ease future debugging, like `cgroup.type <- threaded failed`.